google/resource_google_project_default_service_accounts.go

```diff
@@ -0,0 +1,222 @@
+		if err != nil {
+			return fmt.Errorf("cannot disable service account %s: %v", serviceAccountSelfLink, err)
+		}
+	case "ENABLE":
+		_, err := config.NewIamClient(userAgent).Projects.ServiceAccounts.Enable(serviceAccountSelfLink, &iam.EnableServiceAccountRequest{}).Do()
+		if err != nil {
+			return fmt.Errorf("cannot enable service account %s: %v", serviceAccountSelfLink, err)
+		}
+	case "DEPRIVILEGE":
+		iamPolicy, err := config.NewResourceManagerClient(userAgent).Projects.GetIamPolicy(project, &cloudresourcemanager.GetIamPolicyRequest{
+			Options:         &cloudresourcemanager.GetPolicyOptions{},
+			ForceSendFields: []string{},
+			NullFields:      []string{},
+		}).Do()
+		if err != nil {
+			return fmt.Errorf("cannot get IAM policy on project %s: %v", project, err)
+		}
+
+		for _, bind := range iamPolicy.Bindings {
+			newMembers := []string{}
+			if bind.Role == "roles/editor" {
+				for _, member := range bind.Members {
+					if member != fmt.Sprintf("serviceAccount:%s", email) {
+						newMembers = append(newMembers, member)
+					}
+				}
+			}
+			bind.Members = newMembers
+		}
+		_, err = config.NewResourceManagerClient(userAgent).Projects.SetIamPolicy(project, &cloudresourcemanager.SetIamPolicyRequest{
+			Policy:          iamPolicy,
+			UpdateMask:      "",
+			ForceSendFields: []string{},
+			NullFields:      []string{},
+		}).Do()
+		if err != nil {
+			return fmt.Errorf("cannot update IAM policy on project %s: %v", project, err)
+		}
+	default:
+		return fmt.Errorf("action %s is not a valid action", action)
+	}
+
+	return nil
+}
+
+func resourceGoogleProjectDefaultServiceAccountsCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	userAgent, err := generateUserAgentString(d, config.userAgent)
+	if err != nil {
+		return err
+	}
+	pid, ok := d.Get("project").(string)
+	if !ok {
+		return fmt.Errorf("cannot get project")
+	}
+	action, ok := d.Get("action").(string)
+	if !ok {
+		return fmt.Errorf("cannot get action")
+	}
+
+	serviceAccounts, err := resourceGoogleProjectDefaultServiceAccountsList(config, d, userAgent)
+	if err != nil {
+		return fmt.Errorf("error listing service accounts on project %s: %v", pid, err)
+	}
+	changedServiceAccounts := make(map[string]interface{})
+	for _, sa := range serviceAccounts {
+		// As per documentation https://cloud.google.com/iam/docs/service-accounts#default
+		// we have just two default SAs and the e-mail may change. So, it is been filtered
+		// by the Display Name
+		switch strings.ToLower(sa.DisplayName) {
+		case "compute engine default service account":
+			changedServiceAccounts[sa.UniqueId] = sa.Email
+			err := resourceGoogleProjectDefaultServiceAccountsDoAction(d, meta, action, sa.UniqueId, sa.Email, pid)
+			if err != nil {
+				return fmt.Errorf("error doing action %s on Service Account %s: %v", action, sa.Email, err)
+			}
+		case "app engine default service account":
```

Comment:
Since this logic is exactly the same as the compute engine one (and I imagine if more default SAs came up they would have the same logic as well), this feels more like it should be an if statement (with compute || app engine) instead of a case/switch